runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10350,6 +10350,544 @@ inlineCompareAndSwapNative(
+   TR::Register *xmmMaskReg = cg->allocateRegister(TR_VRF);
+   TR::Register *maskReg = cg->allocateRegister();
+   TR::Register *indexReg = cg->allocateRegister();
+   TR::Register *xmmChunkReg = cg->allocateRegister(TR_VRF);
+   TR::Register *chunkReg = cg->allocateRegister();
+   TR::Register *bytesLeftReg = cg->allocateRegister();
+   TR::Register *resultReg = cg->allocateRegister();
+
+   TR::RegisterDependencyConditions *dependencies = generateRegisterDependencyConditions((uint8_t)9, (uint8_t)9, cg);
+   dependencies->addPreCondition(loopLimitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(limitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(xmmMaskReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(maskReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(indexReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(xmmChunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(chunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(bytesLeftReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(resultReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(loopLimitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(limitReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(xmmMaskReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(maskReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(indexReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(xmmChunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(chunkReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(bytesLeftReg, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(resultReg, TR::RealRegister::NoReg, cg);
+
+   // Labels
+   TR::LabelSymbol *begLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *loopLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *threeOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *fiveOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *nineOrMoreBytesLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualTestLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *returnLenLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *returnIndexLabel = generateLabelSymbol(cg);
+   begLabel->setStartInternalControlFlow();
+   endLabel->setEndInternalControlFlow();
+
+
+   // Beginning label
+   generateLabelInstruction(TR::InstOpCode::label, node, begLabel, cg);
+
+   // index = offset
+   generateRegRegInstruction(TR::InstOpCode::MOV8RegReg, node, indexReg, offsetReg, cg);
+
+   // limit = offset + length
+   generateRegRegInstruction(TR::InstOpCode::MOV8RegReg, node, limitReg, offsetReg, cg);
```

Comment:
You still have added offset before ANDing:

```
loopLimit = ((off + length) & -16) + off
```

It should be as follows:

```
loopLimit = (length & -16) + off
```

Reply:
Hi Brad, I might confused, but I don't think I'm adding offset before ANDing?
```cpp
   // loopLimit = (length & -16) + offset
   generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, loopLimitReg, lengthReg, cg);
   generateRegImmInstruction(TR::InstOpCode::AND4RegImm4, node, loopLimitReg, -16, cg);
   generateRegRegInstruction(TR::InstOpCode::ADD4RegReg, node, loopLimitReg, offsetReg, cg);
```

Reply:
Sorry, I confused limitReg for loopLimitReg